e2e/config/config.go

```diff
@@ -71,8 +71,23 @@ type Deployer struct {
 	Type string `json:"type"`
 	// Description is a human-readable description of the deployer.
 	Description string `json:"description"`
+	Disapp      Disapp `json:"disapp,omitempty"`
 }
 
+type Disapp struct {
+	// RecipeType is the type of the recipe. Can be one of the following:
+	// - "generate" for the test to generate a recipe based on the workload
+	// - "workload" for the test to use a workload provided recipe
+	// - "internal" for the test to use a ramen internal recipe
+	RecipeType string `json:"recipeType"`
+	// RecipeName is the name of the recipe to use.
+	// If RecipeType is "generate", the generated recipe will use this name.
+	// If RecipeType is "workload", the recipeRef in the DRPC will use this name.
+	// If RecipeType is "internal", the recipeRef will use ramenOps ns as the namespace and this name as the name.
```

Comment:
If we use "vm" instead of internal the deployer can configure as ramen-ops/vm-recipe. And we an have other types that can have logic. So we don't need the name. Lets add the type later if we need it.

Reply:
Type was added because we need it for the workload type cases. If type is already added, I would prefer to keep the list of types minimal and expand the list in the names.

Reply:
But name is not needed even for the workload case. Lets not add stuff before we have a use case. Currently we don't have any workload with recipe so we even don't need this type. When we will add one we will have better understanding of the problem.